libs/s25main/drivers/VideoDriverWrapper.cpp

```diff
@@ -493,3 +494,24 @@ bool VideoDriverWrapper::IsFullscreen() const
 {
     return videodriver->IsFullscreen();
 }
+
+float VideoDriverWrapper::getDpiScale() const
+{
+    return videodriver->getDpiScale();
+}
+
+const GuiScale& VideoDriverWrapper::getGuiScale() const
```

Comment:
Maybe it makes more sense to call the struct and hence the getter `GuiScaler`? Because you set an unsigned but get a struct although the name is seemingly the same.   
Just an idea.

Reply:
In my mind, `GuiScale` encapsulates the percentage as well as the derived scaling factors. The fact that it also bundles the operations is incidental and simply for convenience. I wouldn't want to emphasize the operations by calling it `GuiScaler`.

Reply:
It looks like (almost) the only uses of `GuiScale` are those functions. The encapsulated values are only queried by the options screen. So why don't "emphasize the operations" if that's how it is (mainly) used?

And this looks a bit odd to me:
```
const GuiScale& getGuiScale();
void setGuiScale(float);
```

Reply:
> The places that set the GUI scale don't care about the operations and only know the scaling percentage, while the opposite holds for the callers of `getGuiScale()`. IMO, the asymmetric requirements justify the asymmetry in the API.

True, so the first variant is better I guess which leaves open adding `getGuiScale` wrapping `getGuiScaler().getPercent()` if we need that often enough.

Reply:
```sh
$ grep -r 'percent()' extras/ libs/
libs/driver/include/driver/GuiScale.h:    constexpr unsigned percent() const noexcept { return percent_; }
libs/driver/src/VideoDriver.cpp:    if(guiScale_.percent() == percent)
libs/s25main/desktops/dskOptions.cpp:            if(VIDEODRIVER.getGuiScale().percent() == 100
```
Not really.

Reply:
As I'm about to rename it, `GuiScaler` still sounds wrong to me in context. It implies that this class is doing the actual scaling of the UI when that is accomplished by setting the viewport and projection accordingly. The logic that defining operations on `GuiScale` should turn it into `GuiScaler` doesn't quite track when, for example, adding addition operations to `Vector2D` wouldn't turn it into `Vector2DAdder` – it's still just a vector. A class that accepts two or more vectors and adds them would be different.

When I read `mouse_xy.pos = getGuiScaler().screenToView(Position(ev.button.x, ev.button.y));` `GuiScaler` sounds like a much more complicated entity than a simple wrapper of pre-computed values with some operations that are bundled for convenience. `GuiScale` just hides the implementation detail that division by the scaling factor is done by multiplying with the reciprocal. And with the functions, users don't have to think about which direction requires multiplying or dividing by the scaling factor.

Reply:
If it's just the asymmetry in the getter and setter that's bothering you, might I suggest:
```cpp
void setGuiScalePercent(unsigned);
```

Reply:
The whole thing was just a suggestion how the asymmetry might be (less) confusing. So I'm fine with keeping it.

However I like your idea of `setGuiScalePercent`. That is even better as it includes the unit in the name.